core/common/src/main/java/alluxio/network/protocol/databuffer/DataNettyBufferV2.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol.databuffer;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A DataBuffer with the underlying data being a {@link ByteBuf}. This is a new version of
+ * {@link DataNettyBuffer}.
+ */
+ // TODO(peis): Deprecate {@link DataNettyBuffer}.
```

Comment:
The new version is a direct wrapper of netty ByteBuf.

The old version tries to convert netty ByteBuf to java's ByteBuffer.  And it also increases netty data buffer's refcnt.